contracts/EverlongStrategy.sol

```diff
@@ -520,6 +546,183 @@ contract EverlongStrategy is BaseStrategy {
+        }
+    }
+
+    /// @dev Open a long with the specified amount of assets. Return the amount
+    ///      of bonds received and their maturityTime.
+    /// @param _toSpend Amount of strategy assets to spend.
+    /// @param _minOutput Minimum amount of bonds to accept.
+    /// @param _minVaultSharePrice Minimum hyperdrive vault share price to
+    ///                            purchase at.
+    /// @param _extraData Extra data to pass to hyperdrive.
+    /// @return maturityTime Maturity time for bonds received.
+    /// @return bondAmount Amount of bonds received.
+    function _openLong(
+        uint256 _toSpend,
+        uint256 _minOutput,
+        uint256 _minVaultSharePrice,
+        bytes memory _extraData
+    ) internal returns (uint256 maturityTime, uint256 bondAmount) {
+        // Prepare for opening the long differently if the strategy asset is
+        // a wrapped hyperdrive token.
+        if (isWrapped) {
+            // The strategy asset is a wrapped hyperdrive token so it must be
+            // unwrapped.
+            _toSpend = _unwrap(_toSpend);
+
+            // Approve hyperdrive for the unwrapped asset, which is also the
+            // `hyperdriveToken`.
+            //
+            // Leave the approval slot dirty to save gas on future approvals.
+            ERC20(hyperdriveToken).forceApprove(
+                address(hyperdrive),
+                _toSpend + 1
+            );
+
+            // Convert back to hyperdrive's base denomination, same as the
+            // wrapped token's.
+            _toSpend = convertToWrapped(_toSpend);
+        }
+        // The strategy asset is not wrapped, no conversions are necessary.
+        // Approve the hyperdrive contract for strategy asset.
+        //
+        // Leave the approval slot dirty to save gas on future approvals.
+        else {
+            ERC20(asset).forceApprove(address(hyperdrive), _toSpend + 1);
+        }
+
+        // Open the long. Return the maturity time and amount of bonds received.
+        (maturityTime, bondAmount) = IHyperdrive(hyperdrive).openLong(
+            asBase,
+            _toSpend,
+            _minOutput,
+            _minVaultSharePrice,
+            _extraData
+        );
+    }
+
+    /// @dev Preview the amount of bonds received from opening a position with
+    ///      the specified amount of strategy assets.
+    /// @param _toSpend Amount of strategy assets to spend.
+    /// @param _extraData Extra data to pass to hyperdrive.
+    /// @return bondAmount Amount of bonds that would be received.
+    function _previewOpenLong(
+        uint256 _toSpend,
+        bytes memory _extraData
+    ) internal view returns (uint256 bondAmount) {
+        // Call the preview function and return the expected amount of bonds
+        // to be received.
+        bondAmount = IHyperdrive(hyperdrive).previewOpenLong(
+            asBase,
+            _poolConfig,
+            _toSpend,
+            _extraData
+        );
+    }
```

Comment:
It seems like this is more of a nice to have refactor than a requirement. Unless I'm missing something, I'd consider reverting this change to reduce the number of lines that are changed by this PR.

I wouldn't be saying this if we were making this change before the audit, but now that we're in audits, it's best to keep the changes as small as possible.

Reply:
Totally on board with that